### ISSUE TITLE
Annotation Bugs - Control points visibility + Dragged text label's position in PNG/SVG exporting

### ISSUE THREAD

**Issue by nishikantparmariam** [highcharts/highcharts#19971]
_(Note - Open all GIFs in new tab for clarity)_

#### Expected behavior
- PNG/SVG export should respect the position of text label if has been dragged by a control point.
- By default control points should not be activated when annotation is added.

#### Live demo with steps to reproduce
**https://jsfiddle.net/v1mo0bch/**

#### Bugs

- **Bug 1** - PNG/SVG exporting does not seem to respect the position of text label if it has been dragged to some other point, it still shows text label in its original position. This is blocker for anyone using annotation because usually user adds an annotation, interactively adjust its position and expect the downloaded PNG to be WYSIWYG.

  ![text-drag-not-working](https://github.com/highcharts/highcharts/assets/7736252/9d7099e9-9c21-48b7-81df-6668b18fe515) 

  **Partial Workaround** - I found that if I add this line `target.annotation.options.labels[0].point = target.points[0].options;` in text label's `controlPoints`' drag event then PNG exporting works and dragged position is respected.

   This was inspired by this - https://github.com/highcharts/highcharts/blob/65a8cbb2058163c0783883abe8785601580fe7de/ts/Extensions/Annotations/Types/CrookedLine.ts#L245-L249

  **But, this workaround does not work in Bug 2 like cases which I am running into, so its unusable.**

   (This issue also happens in this minimal example from highcharts samples - **https://jsfiddle.net/royt8z2w/**)

- **Bug 2** - When adding an annotation, control points are getting activated by default - probably because `click` event is being triggered. In below example, it is happening only if _line is at certain angle but in my application it is happening always_.
   
    ![by-default-control-point-active](https://github.com/highcharts/highcharts/assets/7736252/e33c1833-9280-4a8c-8aff-a3c526b6b60c)

    How this relates to Bug 1 - 
    
    In below GIF, I've added above workaround so that text label's position is respected upon dragging. But it works only for first annotation (control point **not** activated right after adding ) and not for second annotation (where control point **got activated** right after adding it automatically - Bug 2) - 

   ![partial-fix](https://github.com/highcharts/highcharts/assets/7736252/f9b2fc89-fc14-4e08-a2ea-8ce7d671681b)

  So, workaround is not fully usable.

  Weirdly, if workaround has been added and - 

   - Text label's control point is deactivated first (in Bug 2 case where it got added automatically) by clicking elsewhere
   - **Activated again** and text label is dragged
  
   Then in exported image, position of text label is the dragged position.
   

#### Use-case

My use-case is to sync visibility of a `div` with the annotation. This div will have UI where users can edit the annotation. I want to show control points (and my div) right after annotation has been added (**so Bug 2 is actually a desired behavior for me**), but Bug 1 is a blocker. 

Additional question - 
- Should I write a plugin myself that wraps `setControlPointsVisibiliy` and toggle the `div` based on annotation's control point is active or not? 
- I noticed that if control points are active, then they are deactivated only if user clicks somewhere else on the **chart**, is it possible to deactivate the control points if user clicks anywhere else on the screen including chart, should I attach events to `document` myself?

#### Product version
Highcharts 10.3.3

#### Affected browser(s)
Chrome

cc: @pawelfus @karolkolodziej would be really helpful if you can provide quick workarounds, TIA!

**Comment by nishikantparmariam** [comment]
Thanks for the quick turnaround @pawelfus!

> I'm guessing this, because updating `chart.options.annotations[...]` works fine

This workaround seems to resolve the exporting issue. Thanks!

> There is also an alternative approach: instead of `translatePoint()` use `translate()`

Yes, I had tried `translate` also that does not has exporting issues, but my use-case is to drag the point itself and not change the offset around point.

> At this moment we don't have the option to change whether the annotation is unselected upon clicking on the chart or on document/body

Ohk, I'll try to implement this myself as needed.

**Comment by nishikantparmariam** [comment]
@pawelfus, I noticed there is an issue with a user programmatically calling `annotation.setControlPointsVisibility(true)` - this makes control points visible but clicking anywhere else on the chart does not hide the control point (which is the usual behavior).

After checking the NavigationBindings codebase - 

https://github.com/highcharts/highcharts/blob/b822429fa0ed7c16e4e5211b65d9a41beffd77bc/ts/Extensions/Annotations/NavigationBindings.ts#L718-L731

I found workaround that one must set - 

```js
annotation.chart.navigationBindings.activeAnnotation = annotation;
```

Reproducer - https://jsfiddle.net/he03wouc/

-----

Also, setting `annotation.setControlPointsVisibility(true)` does not trigger `showPopup` event, so as workaround, I can call it myself - `annotation.chart.navigationBindings.options.events.showPopup.call(...)`

Wanted to check if this looks good.

**Comment by pawelfus** [comment]
> I noticed there is an issue with a user programmatically calling annotation.setControlPointsVisibility(true) - this makes control points visible but clicking anywhere else on the chart does not hide the control point (which is the usual behavior).

That's true. I noticed that too, but didn't mention it since you said you want custom logic to hide these points, thus you will need to use `.setControlPointsVisibility(false)` anyway.

___

> Also, setting annotation.setControlPointsVisibility(true) does not trigger showPopup event, so as workaround, I can call it myself - annotation.chart.navigationBindings.options.events.showPopup.call(...)
>
> Wanted to check if this looks good.

Correct again 👍 

Note: In addition to the above, you need to hide the popup when clicking on the other elements on your website. We fire `closePopup` only on chart click.
